src/hooks/usePricing.ts

```diff
@@ -0,0 +1,89 @@
+import { currencyService } from '@/components/services/currency.service';
+import { ProductsProps, stripeService } from '@/components/services/stripe.service';
+import { useEffect, useReducer } from 'react';
+import { CouponType } from '@/pages/api/stripe/get_coupons';
+
+interface UseStripeProductsAndCurrencyResponse {
+  products: ProductsProps | undefined;
+  loadingCards: boolean;
+  currency: string;
+  coupon: CouponType | null;
+}
+
+type ActionType =
+  | { type: 'SET_PRODUCTS'; payload: ProductsProps | undefined }
+  | { type: 'SET_LOADING_CARDS'; payload: boolean }
+  | { type: 'SET_CURRENCY'; payload: string }
+  | { type: 'SET_COUPON'; payload: CouponType | null };
+
+const reducer = (state: any, action: ActionType) => {
+  switch (action.type) {
+    case 'SET_PRODUCTS':
+      return { ...state, products: action.payload };
+    case 'SET_LOADING_CARDS':
+      return { ...state, loadingCards: action.payload };
+    case 'SET_CURRENCY':
+      return { ...state, currency: action.payload };
+    case 'SET_COUPON':
+      return { ...state, coupon: action.payload };
+    default:
+      return state;
+  }
+};
+
+function usePricing(couponCode?: CouponType): UseStripeProductsAndCurrencyResponse {
+  const initialState = {
+    products: undefined,
+    loadingCards: true,
+    currency: '€',
+    coupon: null,
+  };
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const fetchData = async () => {
+    try {
+      const prices = await stripeService.getPrices();
+      dispatch({ type: 'SET_PRODUCTS', payload: prices });
+      dispatch({ type: 'SET_LOADING_CARDS', payload: false });
+    } catch (error) {
+      try {
+        const res = await stripeService.getPrices(true);
+        dispatch({ type: 'SET_PRODUCTS', payload: res });
+        dispatch({ type: 'SET_LOADING_CARDS', payload: false });
+      } catch (error) {
+        console.error('Error getting prices');
```

Comment:
No user is going to check the console for errors, propagate the error in some way so the UI can display the error in case is needed

`setError(error)` in the hook and return the error will do the trick